sklearn/preprocessing/mice.py

```diff
@@ -0,0 +1,365 @@
+        missing_values will be imputed. For missing values encoded as
+        np.nan, use the string value "NaN".
+
+    imputation_order : str, optional, default="monotone"
+        The order in which the columns will be imputed.
+        "monotone" - From columns with fewest missing values to most.
+        "revmonotone" - From columns with most missing values to least.
+        "roman" - Left to right.
+        "arabic" - Right to left.
+        "random" - A random order for each round.
+
+    n_imputations : int, optional, defaults = 100.
+        Number of MICE rounds to perform the results of which will be
+        used in the final average.
+
+    n_burn_in : int, optional, default = 10.
+        Number of initial MICE rounds to perform the results of which
+        will not be returned.
+
+    n_nearest_columns : int, optional, default = np.infty.
+        Number of other columns to use to estimate the missing values of
+        the current column. Can provide significant speed-up
+        when the number of columns is huge.
+
+    initial_fill_method : str, optional, default = "mean"
+        Valid values: {"mean", "median", or "most_frequent"}.
+        Uses sklearn.preprocessing.Imputer.
+
+    min_value : float, default = None
+        Minimum possible imputed value.
+
+    max_value : float, default = None
+        Maximum possible imputed value.
+
+    verbose : boolean, default = False
+        Whether to provide updates on progress.
+
+    Notes
+    -----
+    - Columns which only contain missing values at `fit` are discarded upon
+        `transform`.
+
+    References
+    ----------
+
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            n_nearest_columns=np.infty,
+            initial_fill_method="mean",
+            min_value=None,
+            max_value=None,
+            verbose=False):
+        from ..linear_model import BayesianRidge  # avoiding circular import
+        self.model = BayesianRidge()
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.n_nearest_columns = n_nearest_columns
+        self.initial_fill_method = initial_fill_method
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.trained_model_triplets = []
+
+    def _fill_in_one_column(self,
+                            X_filled,
+                            mask_missing_values,
+                            this_column,
```

Comment:
I am not the good guy to criticize the variable naming, but I don't think this is right. `feature_idx` or something like that could be more meaningful probably?